Objects/unicodeobject.c

```diff
@@ -4700,6 +4700,8 @@ static Py_ssize_t
 ascii_decode(const char *start, const char *end, Py_UCS1 *dest)
 {
     const char *p = start;
+    const char *size_t_end  = end - SIZEOF_SIZE_T;
+    const char *unrolled_end = end - (4 * SIZEOF_SIZE_T);
 
 #if SIZEOF_SIZE_T <= SIZEOF_VOID_P
```

Comment:
Is CPython supported on some platform where `size_t` is strictly greater than `void*`? I know this PR doesn't touch this conditional, but I'm curious. @encukou

Reply:
I'm not aware of any platform where `sizeof(size_t) != sizeof(void*)`.

Reply:
This is not really about supported platforms (i.e. where *we* run the tests), or *currently* supported platforms.
We should aim for standard C whenever it's reasonable; if a speedup needs an extra assumption it should document that with this kind of `#if`.